libraries/ESP8266mDNS/src/LEAmDNS2Host_Structs.cpp

```diff
@@ -0,0 +1,3187 @@
+            else
+            {
+                break;
+            }
+        }
+    }
+    return bResult;
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::c_str
+
+*/
+const char* clsLEAMDNSHost::clsServiceTxts::c_str(void) const
+{
+
+    if ((!m_pcCache) &&
+            (m_Txts.size()) &&
+            ((((clsServiceTxts*)this)->m_pcCache = new char[c_strLength()])))   // TRANSPARENT caching
+    {
+        ((clsServiceTxts*)this)->c_str(m_pcCache);
+    }
+    return m_pcCache;
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::bufferLength
+
+    (incl. closing '\0').
+
+*/
+size_t clsLEAMDNSHost::clsServiceTxts::bufferLength(void) const
+{
+    return (length() + 1);
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::buffer
+
+*/
+bool clsLEAMDNSHost::clsServiceTxts::buffer(char* p_pcBuffer)
+{
+    bool bResult = false;
+
+    if (p_pcBuffer)
+    {
+        bResult = true;
+
+        *p_pcBuffer = 0;
+        for (const clsServiceTxt* pTxt : m_Txts)
+        {
+            *(unsigned char*)p_pcBuffer++ = pTxt->length();
+            size_t  stLength;
+            if ((bResult = (0 != (stLength = (pTxt->m_pcKey ? strlen(pTxt->m_pcKey) : 0)))))
+            {
+                strcpy(p_pcBuffer, pTxt->m_pcKey);
+                p_pcBuffer += stLength;
+                *p_pcBuffer++ = '=';
+                if ((stLength = (pTxt->m_pcValue ? strlen(pTxt->m_pcValue) : 0)))
+                {
+                    strcpy(p_pcBuffer, pTxt->m_pcValue);
+                    p_pcBuffer += stLength;
+                }
+            }
+            else
+            {
+                break;
+            }
+        }
+    }
+    return bResult;
+}
+
+/*
+    clsLEAMDNSHost::clsServiceTxts::compare
+
+*/
+bool clsLEAMDNSHost::clsServiceTxts::compare(const clsLEAMDNSHost::clsServiceTxts& p_Other) const
+{
+    bool    bResult = false;
+
+    if ((bResult = (length() == p_Other.length())))
+    {
+        // Compare A->B
+        for (const clsServiceTxt* pTxt : m_Txts)
+        {
+            const clsServiceTxt*    pOtherTxt = p_Other.find(pTxt->m_pcKey);
+            if (!((bResult = ((pOtherTxt) &&
+                              (pTxt->m_pcValue) &&
+                              (pOtherTxt->m_pcValue) &&
+                              (strlen(pTxt->m_pcValue) == strlen(pOtherTxt->m_pcValue)) &&
+                              (0 == strcmp(pTxt->m_pcValue, pOtherTxt->m_pcValue))))))
+            {
+                break;
+            }
+        }
+        // Compare B->A
+        for (const clsServiceTxt* pOtherTxt : p_Other.m_Txts)
+        {
+            const clsServiceTxt*    pTxt = find(pOtherTxt->m_pcKey);
+            if (!((bResult = ((pTxt) &&
+                              (pOtherTxt->m_pcValue) &&
+                              (pTxt->m_pcValue) &&
+                              (strlen(pOtherTxt->m_pcValue) == strlen(pTxt->m_pcValue)) /* &&
+                              redundant: (0 == strcmp(pOtherTxt->m_pcValue, pTxt->m_pcValue)) */))))
```

Comment:
This is backwards.  The strlen can be removed, because we just want to see if both strings are identical.  So anywhere this kind of change was done, need to swap...

Reply:
Fixed